draft-ietf-core-dns-over-coap.md

```diff
@@ -320,71 +329,91 @@ the Content-Format option by the DoC server.
 
-The RECOMMENDED algorithm to assure the requirement for the DoC is to set the Max-Age option of a response to the minimum TTL of a DNS response and to subtract this value from all TTLs of that DNS response.
+The RECOMMENDED algorithm to meet the requirement for DoC is as follows:
+Set the Max-Age option of a response to the minimum TTL of a DNS response and subtract this value from all TTLs of that DNS response.
 This prevents expired records unintentionally being served from an intermediate CoAP cache.
 Additionally, it allows for the ETag value for cache validation, if it is based on the content of the response, not to change even if the TTL values are updated by an upstream DNS cache.
 If only one record set per DNS response is assumed, a simplification of this algorithm is to just set all TTLs in the response to 0 and set the TTLs at the DoC client to the value of the Max-Age option.
 
-### DNS Update {#sec:dns-update}
+If shorter caching periods are plausable, e.g., if the RCODE of the message indicates an error that should only be cached for a minimal duration, the value for the Max-Age option SHOULD be set accordingly.
+This value might be 0, but if the DoC server knows the error will persist, greater values are also conceivable, depending on the projected duration of the error.
+Same goes for DNS responses, that for any reason do not carry any records with a TTL.
 
-Until future work provides considerations for DNS Update {{-dns-update}}, a DoC server that receives a query with the UPDATE opcode SHOULD indicate in its response that it does not implement DNS Update, see {{-dns-update}}.
+### Examples {#sec:resp-examples}
 
-### Examples
-
-The following examples illustrate the replies to the query "example.org. IN
+The following example illustrates the response to the query "example.org. IN
 AAAA record", recursion turned on. Successful responses carry one answer
-record including address 2001:db8:1::1:2:3:4 and TTL 58719.
+record including address 2001:db8:1::1:2:3:4 and TTL 79689.
+As in {{sec:req-examples}}, we use both a human-readable and binary form of the payload.
 
 A successful response:
 
     2.05 Content
     Content-Format: application/dns-message
     Max-Age: 58719
-    Payload: 00 00 81 a0 00 01 00 01 00 00 00 00 07 65 78 61 [binary]
-             6d 70 6c 65 03 6f 72 67 00 00 1c 00 01 c0 0c 00 [binary]
-             1c 00 01 00 01 37 49 00 10 20 01 0d b8 00 01 00 [binary]
-             00 00 01 00 02 00 03 00 04                      [binary]
+    Payload (human-readable):
+      ;; ->>Header<<- opcode: QUERY, status: NOERROR, id: 0
+      ;; flags: qr rd ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ARCOUNT: 0
+
+      ;; QUESTION SECTION:
+      ;example.org.                 IN      AAAA
+      ;; ANSWER SECTION:
+      ;example.org.         79689   IN      AAAA    2001:db8:1::1:2:3:4
 
-When a DNS error (SERVFAIL in this case) is noted in the DNS response, the CoAP
-response still indicates success:
+    Payload (binary):
```

Comment:
Not sure (and @waehlisch backen me about this offline) if these additional binary is more confusing than helpful. What do you think, @cabo?

Reply:
I slept on it and I decided to remove them.

Reply:
No, I think as well that they are more useful for the examples than the hex dumps. I am just worried people would put this into their CoAP messages. But I hope, the sentence above and the parathesis `(human-readable)` make it abundantly clear that this is not the actual message format.